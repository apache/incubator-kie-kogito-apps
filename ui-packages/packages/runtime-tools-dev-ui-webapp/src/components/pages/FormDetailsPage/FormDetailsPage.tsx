/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useEffect, useState } from 'react';
import {
  Card,
  Label,
  PageSection,
  Text,
  TextVariants
} from '@patternfly/react-core';
import { OUIAProps, ouiaPageTypeAndObjectId } from '@kogito-apps/ouia-tools';
import FormDetailsContainer from '../../containers/FormDetailsContainer/FormDetailsContainer';
import '../../styles.css';
import { useHistory } from 'react-router-dom';
import { FormInfo } from '@kogito-apps/forms-list';
import { PageTitle } from '@kogito-apps/consoles-common';
import Moment from 'react-moment';
import FormNotification, {
  Notification
} from '../TaskDetailsPage/components/FormNotification/FormNotification';

const FormDetailsPage: React.FC<OUIAProps> = () => {
  const [notification, setNotification] = useState<Notification>();

  useEffect(() => {
    return ouiaPageTypeAndObjectId('form-detail');
  });
  const history = useHistory();
  const formData: FormInfo = history.location.state['formData'];

  const onSuccess = () => {
    const message = `The form '${formData.name}.${formData.type}' has been successfully saved.`;

    showNotification('success', message);
  };

  const onError = (details?: string) => {
    const message = `The form '${formData.name}.${formData.type}' couldn't be saved.`;

    showNotification('error', message, details);
  };

  const showNotification = (
    notificationType: 'error' | 'success',
    submitMessage: string,
    notificationDetails?: string
  ) => {
    setNotification({
      type: notificationType,
      message: submitMessage,
      details: notificationDetails,
      close: () => {
        setNotification(null);
      }
    });
  };

  return (
    <React.Fragment>
      <PageSection variant="light">
        <PageTitle
          title={formData.name}
          extra={<Label variant="outline">{formData.type}</Label>}
        />
        <Text component={TextVariants.p} style={{ marginTop: '10px' }}>
          <span style={{ fontWeight: 'bold' }}>Last modified:</span>{' '}
          <Moment fromNow>{formData.lastModified}</Moment>
        </Text>
        {notification && (
          <div className="kogito-task-console__task-details-page">
            <FormNotification notification={notification} />
          </div>
        )}
      </PageSection>
      <PageSection>
        <Card className="Dev-ui__card-size">
          <FormDetailsContainer
            formData={formData}
            onSuccess={onSuccess}
            onError={onError}
          />
        </Card>
      </PageSection>
    </React.Fragment>
  );
};

export default FormDetailsPage;
