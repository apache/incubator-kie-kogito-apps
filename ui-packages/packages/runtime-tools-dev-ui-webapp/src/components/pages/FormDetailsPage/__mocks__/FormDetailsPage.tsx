import React from 'react';

const MockedFormDetailsPage: React.FC = () => {
  return <></>;
};

export default MockedFormDetailsPage;
