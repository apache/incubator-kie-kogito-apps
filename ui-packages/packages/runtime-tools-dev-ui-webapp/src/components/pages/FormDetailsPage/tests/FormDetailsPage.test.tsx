/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { mount } from 'enzyme';
import FormDetailsPage from '../FormDetailsPage';
import { BrowserRouter } from 'react-router-dom';

jest.mock('../../../containers/FormDetailsContainer/FormDetailsContainer');

Date.now = jest.fn(() => 1592000000000); // UTC Fri Jun 12 2020 22:13:20

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    location: {
      state: {
        formData: {
          name: 'form1',
          type: 'html',
          lastModified: new Date('2021-08-23T13:26:02.13Z')
        }
      }
    }
  })
}));

describe('FormDetailsPage tests', () => {
  it('Snapshot', () => {
    const wrapper = mount(
      <BrowserRouter>
        <FormDetailsPage />
      </BrowserRouter>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MockedFormDetailsContainer').exists()).toBeTruthy();
  });
});
