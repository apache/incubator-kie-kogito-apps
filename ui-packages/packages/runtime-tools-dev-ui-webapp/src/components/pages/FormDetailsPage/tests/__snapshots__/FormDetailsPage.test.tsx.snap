// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormDetailsPage tests Snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <FormDetailsPage>
      <PageSection
        variant="light"
      >
        <section
          className="pf-c-page__main-section pf-m-light"
        >
          <PageTitle
            extra={
              <Label
                variant="outline"
              >
                html
              </Label>
            }
            title="form1"
          >
            <Flex
              data-ouia-component-type="page-title"
              data-ouia-safe={true}
            >
              <div
                className="pf-l-flex"
                data-ouia-component-type="page-title"
                data-ouia-safe={true}
              >
                <FlexItem
                  spacer={
                    Object {
                      "default": "spacerSm",
                    }
                  }
                >
                  <div
                    className="pf-m-spacer-sm"
                  >
                    <Title
                      headingLevel="h1"
                      size="4xl"
                    >
                      <h1
                        className="pf-c-title pf-m-4xl"
                      >
                        form1
                      </h1>
                    </Title>
                  </div>
                </FlexItem>
                <FlexItem
                  spacer={
                    Object {
                      "default": "spacerSm",
                    }
                  }
                >
                  <div
                    className="pf-m-spacer-sm"
                  >
                    <Label
                      variant="outline"
                    >
                      <span
                        className="pf-c-label pf-m-outline"
                      >
                        <span
                          className="pf-c-label__content"
                        >
                          html
                        </span>
                      </span>
                    </Label>
                  </div>
                </FlexItem>
              </div>
            </Flex>
          </PageTitle>
          <Text
            component="p"
            style={
              Object {
                "marginTop": "10px",
              }
            }
          >
            <p
              className=""
              data-ouia-component-id="OUIA-Generated-Text-1"
              data-ouia-component-type="PF4/Text"
              data-ouia-safe={true}
              data-pf-content={true}
              style={
                Object {
                  "marginTop": "10px",
                }
              }
            >
              <span
                style={
                  Object {
                    "fontWeight": "bold",
                  }
                }
              >
                Last modified:
              </span>
               
              <t
                ago={false}
                calendar={false}
                decimal={false}
                element={null}
                filter={[Function]}
                fromNow={true}
                interval={60000}
                local={false}
                onChange={[Function]}
                titleFormat=""
                toNow={false}
                unit={null}
                unix={false}
                utc={false}
                withTitle={false}
              >
                <time
                  dateTime={"2021-08-23T13:26:02.130Z"}
                >
                  in a year
                </time>
              </t>
            </p>
          </Text>
        </section>
      </PageSection>
      <PageSection>
        <section
          className="pf-c-page__main-section"
        >
          <Card
            className="Dev-ui__card-size"
          >
            <article
              className="pf-c-card Dev-ui__card-size"
              data-ouia-component-id="OUIA-Generated-Card-1"
              data-ouia-component-type="PF4/Card"
              data-ouia-safe={true}
              id=""
            >
              <MockedFormDetailsContainer
                formData={
                  Object {
                    "lastModified": 2021-08-23T13:26:02.130Z,
                    "name": "form1",
                    "type": "html",
                  }
                }
                onError={[Function]}
                onSuccess={[Function]}
              />
            </article>
          </Card>
        </section>
      </PageSection>
    </FormDetailsPage>
  </Router>
</BrowserRouter>
`;
